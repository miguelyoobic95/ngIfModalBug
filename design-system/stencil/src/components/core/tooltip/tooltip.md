---
name: Tooltip
category: Core Components
---

The tooltip component is used with the `yoo-tooltip` tag. The text to be in the tooltip needs to be specified and additional options may be set to modify properties of the tooltip.
Any content may be inserted inside the `<yoo-tooltip></yoo-tooltip>` and the tooltip will appear around this content. 

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`text`|string|  |the text inside the tooltip|
|`placement`|string|  |the location of the tooltip relative to the inner content (top, bottom, left, right, top-start, top-end)|
|`options`|any|  |key value pairs of additional tooltip options|

```yoo-tooltip.html
    <br>
    <br>
    <yoo-tooltip text="I am a top placed tooltip" placement="top">
        <div>Any content can be in this slot</div>
    </yoo-tooltip>
    <br>
    <br>
    <yoo-tooltip id="tooltip2" text="I am a left tooltip" placement="left">
        <div>Any content can be here</div>
    </yoo-tooltip>
    <br>
    <br>
    <br>
    <yoo-tooltip text="I am a right tooltip" placement="right">
        <div>Any content can be in this slot</div>
    </yoo-tooltip>
    <br>
    <br>
    <yoo-tooltip text="I am a bottom placed tooltip" placement="bottom">
        <div>Any content can be in this slot</div>
    </yoo-tooltip>
    <br>
    <br>
```

```yoo-tooltip.js
    var cmp = document.querySelector('yoo-tooltip');
    cmp.options = {};
```

```yoo-tooltip.css hidden
    yoo-tooltip {
        width: 100px;
    }
    #tooltip2 {
        float: right;
    }
```

## Size Variation

Differently sized tooltips may be specified in the options. Size values can be: `small`, `regular`, `large`.

```yoo-tooltip-size.html
    <yoo-tooltip id="tooltip1" text="I am a small right placed tooltip" placement="right">
        <div>Any content can be in this slot</div>
    </yoo-tooltip>
    <br>
    <br>
    <yoo-tooltip id="tooltip2" text="I am a regular bottom placed tooltip" placement="bottom">
        <div>Any content can be in this slot</div>
    </yoo-tooltip>
    <br><br>
     <yoo-tooltip id="tooltip3" text="I am a large top placed tooltip" placement="top">
        <div>Any content can be in this slot</div>
    </yoo-tooltip>
```

```yoo-tooltip-size.js
    var cmp1 = document.querySelector('#tooltip1');
    cmp1.options = { size: "small" };
    var cmp2 = document.querySelector('#tooltip2');
    cmp2.options = { size: "regular" };
    var cmp3 = document.querySelector('#tooltip3');
    cmp3.options = { size: "large" };
```

```yoo-tooltip-size.css hidden
    yoo-tooltip {
        width: 100px;
    }
```