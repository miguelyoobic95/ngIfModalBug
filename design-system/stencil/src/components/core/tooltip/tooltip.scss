

/deep/ {
    @import "../../../../../../node_modules/tippy.js/dist/themes/light";
}

:host {
    display: block;
    cursor: pointer;
}