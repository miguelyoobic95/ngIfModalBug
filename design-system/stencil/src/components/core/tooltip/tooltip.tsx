import { Component, Prop, Element, Watch } from '@stencil/core';
import tippy from 'tippy.js';

@Component({
    tag: 'yoo-tooltip',
    styleUrl: 'tooltip.scss',
    scoped: true
})
export class YooTooltipComponent {
    @Prop() placement: 'top' | 'bottom' | 'left' | 'right' | 'top-start' | 'top-end';
    @Prop() options: any;
    @Prop() text: string;

    @Element() protected host: HTMLElement;

    protected tippy: any = null;

    @Watch('options')
    onOptionsChange() {
        this.loadTippy();
    }

    @Watch('text')
    onTextChange() {
        this.loadTippy();
    }

    componentWillLoad() {
        this.loadTippy();
    }

    loadTippy(): void {
        // Title is not an option in the API; set it directly as an attribute
        this.host.setAttribute('title', this.text);
        let tippyOptions = {
            theme: 'light',
            placement: this.placement,
            arrow: true,
            arrowType: 'round',
            createPopperInstanceOnInit: false,
            ...this.options
        };
        if (this.tippy) {
            this.tippy.destroyAll();
        }
        // Moved tooltip to always be on the host element
        this.tippy = tippy(this.host, tippyOptions);
    }

    render(): JSX.Element {
        return <slot></slot>;
    }
}
