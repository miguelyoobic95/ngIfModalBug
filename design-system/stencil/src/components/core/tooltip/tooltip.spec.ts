jest.mock('tippy.js');

import { TestWindow } from '@stencil/core/testing';
import { YooTooltipComponent } from './tooltip';
// import tippy from 'tippy.js';

describe('YooTooltipComponent', () => {
    let window;
    beforeEach(() => {
        window = new TestWindow();
    });
    it('should build', () => {
        expect(new YooTooltipComponent()).toBeTruthy();
    });

    describe('property rendering', () => {
        it ('should render a basic tooltip with text', async () => {
            let element;
            element = await window.load({
                components: [YooTooltipComponent],
                html: '<yoo-tooltip text="this is the title"></yoo-tooltip>'
            });
            expect(element).toBeTruthy();
            expect(element).toMatchSnapshot();
        });
        it ('should render a basic tooltip with text around some content', async () => {
            let element;
            element = await window.load({
                components: [YooTooltipComponent],
                html: '<yoo-tooltip text="this is the title"><div>Some content</div></yoo-tooltip>'
            });
            expect(element).toMatchSnapshot();
        });
        it ('should render a basic tooltip with text with placement around some content', async () => {
            let element;
            element = await window.load({
                components: [YooTooltipComponent],
                html: '<yoo-tooltip placement="right" text="this is the title"><div>Some content</div></yoo-tooltip>'
            });
            expect(element).toMatchSnapshot();
        });
    });
    describe('setup functionality', () => {
        it ('should call the load tippy function when it renders', async () => {
            let element = new YooTooltipComponent();
            element.loadTippy = jest.fn();
            await element.componentWillLoad();
            expect(element.loadTippy).toHaveBeenCalled();
        });
    });
});