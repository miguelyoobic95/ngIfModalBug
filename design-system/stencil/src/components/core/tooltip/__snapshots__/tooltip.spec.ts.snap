// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooTooltipComponent property rendering should render a basic tooltip with text 1`] = `
<yoo-tooltip
  class="hydrated"
  data-yoo-tooltip-host=""
  data-yoo-tooltip-slot=""
  text="this is the title"
  title="this is the title"
>
  
  
</yoo-tooltip>
`;

exports[`YooTooltipComponent property rendering should render a basic tooltip with text around some content 1`] = `
<yoo-tooltip
  class="hydrated"
  data-yoo-tooltip-host=""
  data-yoo-tooltip-slot=""
  text="this is the title"
  title="this is the title"
>
  
  
  
  <div>
    Some content
  </div>
</yoo-tooltip>
`;

exports[`YooTooltipComponent property rendering should render a basic tooltip with text with placement around some content 1`] = `
<yoo-tooltip
  class="hydrated"
  data-yoo-tooltip-host=""
  data-yoo-tooltip-slot=""
  placement="right"
  text="this is the title"
  title="this is the title"
>
  
  
  
  <div>
    Some content
  </div>
</yoo-tooltip>
`;
